lib/Sema/CSApply.cpp

```diff
@@ -6748,11 +6751,62 @@ Expr *ExprRewriter::convertLiteralInPlace(Expr *literal,
   return literal;
 }
 
-// Resolve @dynamicCallable applications.
+// Returns true if the given method and method type are a valid
+// `@dynamicCallable` required `func dynamicallyCall` method.
+static bool isValidDynamicallyCallMethod(FuncDecl *method,
+                                         AnyFunctionType *methodType) {
```

Comment:
Maybe a better name would be `isValidForDynamicCall` or similar?

Reply:
I don't believe "dynamic call" is a term of art, so I would prefer a name that involves "dynamicCallable" or "dynamicallyCallMethod".

How about `isValidDynamicCallableMethod`?

Reply:
Sure, that works too.

Reply:
Done in https://github.com/apple/swift/pull/24299/commits/cebbd6335dab7e0bbf3e5c0dff9c445588e01f81.